stuffer/s2n_stuffer_pem.c

```diff
@@ -36,11 +37,15 @@
 static int s2n_stuffer_pem_read_encapsulation_line(struct s2n_stuffer *pem, const char *encap_marker,
         const char *keyword)
 {
-    /* Skip any number of Chars until a "-" is reached */
-    POSIX_GUARD(s2n_stuffer_skip_to_char(pem, S2N_PEM_DELIMTER_CHAR));
+    /* Skip any number of Chars until a "--" is reached.
+     * We use "--" instead of "-" to account for dashes that appear in comments.
+     * We do not accept comments that contain "--".
+     */
+    POSIX_GUARD(s2n_stuffer_skip_read_until(pem, S2N_PEM_DELIMITER_TOKEN));
+    POSIX_GUARD(s2n_stuffer_rewind_read(pem, strlen(S2N_PEM_DELIMITER_TOKEN)));
```

Comment:
I think not supporting certificates with 1 dash ("-BEGIN CERTIFICATE-") seems fine. The only concern I have is that any existing certificates loaded like this will be silently skipped rather than fail.

I'm not sure if this is worth doing, but could we maybe partially implement your proposal of searching for "-BEGIN" but only so we can reject suspicious PEMs? Like, I wonder if we could continue searching for a single "-", but branch after a "-" is found. If the next character is also a "-", it's a valid cert so we parse it. But if the next sequence is "BEGIN CERTIFICATE", we fail. Otherwise, we keep searching. Then the PEM would have to be malformed in multiple ways (a single "-" and misspelled) in order to be skipped.

Reply:
If the next character is "BEGIN CERTIFICATE", we would probably just want to accept that as the beginning of a certificate, the way we do now. An error seems unnecessary.

But my concern is the complexity that adds. It changes the parsing strategy, rather than just changing what string we look for. Like, our parsing currently doesn't branch at all. And I don't see much of a difference between ignoring "BEGIN CERTIFICATE" and ignoring "-BEGIN CERTIFICATE-". The correct string is "-----BEGIN CERTIFICATE-----", so I'd be more worried about "----" or "------" than "-". 

Thoughts?